### PR TITLE
[Instrumentation.AWS] Replace net452 target with net462

### LINKED ISSUE #1078: Consider removing `net452` target from the package
# Issue with OpenTelemetry.Contrib.Instrumentation.AWS
The package currently targets `net452` which is a non-supported version. Consider updating it to latest supported version i.e. `net462`.

## Additional Context
This issue is opened as part of #1031 effort

### PR DESCRIPTION
**Pull request by wjrogers**
Fixes #1078.

## Changes

Update OpenTelemetry.Contrib.Instrumentation.AWS's out-of-support `net452` target to `net462`, i.e. `NetFrameworkMinimumSupportedVersion`.

- Update conditional compilation symbols in tests project(s) from NET452 to NETFRAMEWORK.
- Remove now-unnecessary reference to System.Net.Http from tests project.
- Fix a code style warning in the tests project.

### CODE REVIEW

File: src/OpenTelemetry.Contrib.Instrumentation.AWS/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Raised the minimum .NET version to `net462`
```
Comment (cijothomas) [inline src/OpenTelemetry.Contrib.Instrumentation.AWS/CHANGELOG.md:5]: Nit: alternate suggestion:
"Removes .NET Framework 4.5.2, .NET 4.6 support. The minimum .NET Framework version supported is .NET 4.6.2. "

Comment (wjrogers) [reply src/OpenTelemetry.Contrib.Instrumentation.AWS/CHANGELOG.md:5]: I copied the phrasing from
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/b06a6f290fc59a292994e2ba9973eefe0691fa19/src/OpenTelemetry.Contrib.Extensions.AWSXRay/CHANGELOG.md?plain=1#L8-L9

Should I still change it to your suggested phrasing?

Comment (cijothomas) [reply src/OpenTelemetry.Contrib.Instrumentation.AWS/CHANGELOG.md:5]: No! I copied from https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/CHANGELOG.md#130-beta1 and felt it slightly better. But it is very subjective, and it should not block this PR!

**Review by Kielek — APPROVED** [review]

**Comment by cijothomas** [comment]
@srprash Could you review?

**Pull request merged**